src/components/tools/geometry-tool/line-dialog.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from "react";
+import { Button, Dialog } from "@blueprintjs/core";
+import { getBoundingBoxIntersections, solveForY } from "../../../models/tools/geometry/jxg-movable-line";
+import { round } from "lodash";
+
+interface IProps {
+  line: JXG.Line;
+  isOpen: boolean;
+  onAccept: (line: JXG.Line, point1: [number, number], point2: [number, number]) => void;
+  onClose: () => void;
+}
+
+interface IState {
+  slope: string;
+  intercept: string;
+}
+
+export default
+class LineDialog extends React.Component<IProps, IState> {
```

Comment:
I think we should be consistent about using "MovableLine" for code that is specific to the movable line, and reserve "Line" for code that could reasonably be applied to any line. That said, I leave it to your discretion which category this and the other uses fall into.